spring-boot-project/spring-boot/src/test/java/org/springframework/boot/web/server/SslUtilsTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.web.server;
+
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.when;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for {@link SslUtils}.
+ *
+ * @author Chris Bono
+ */
+
+public class SslUtilsTest {
+
+	@Test
+	public void assertStoreContainsAliasPassesWhenAliasFound() throws KeyStoreException {
+		KeyStore keyStore = mock(KeyStore.class);
```

Comment:
So I chose to go w/ mocks on the key store. One interesting point, `containsAlias` is final. I got around it easy w/ `mock-maker-inline`. If there are strong opinions against this approach I could just load the `test.jks` store instead. However, I would not be able to test the "throws exception during call to containsAlias" flow that way.

Reply:
`containsAlias` will throw an exception if you call it on a `KeyStore` that hasn't been initialised. I think you could test the exception handling with the following:

```
KeyStore keyStore = KeyStore.getInstance("JKS");
assertThatThrownBy(() -> SslUtils.assertStoreContainsAlias(keyStore, "alias"))
 				.isInstanceOf(IllegalStateException.class)
 				.hasMessage("Could not determine if keystore contains alias 'alias'").hasCause(keyStoreEx);
```

Reply:
Ah good point @wilkinsona . What is your opinion/preference here - Mockito or load actual test KeyStore?

Reply:
As the alternative is to change the mock maker for the whole project, I'd prefer to load an actual `KeyStore`.

Reply:
Yep this cross my mind too - a wide change for a teeny tiny test case - but figured I would get a 2nd opinion. I will adjust it sometime today.

Reply:
This has ben removed @wilkinsona